packages/shorebird_cli/lib/src/commands/release_new/release_new_command.dart

```diff
@@ -0,0 +1,420 @@
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+import 'package:shorebird_code_push_protocol/shorebird_code_push_protocol.dart';
+
+typedef ResolveReleaser = Releaser Function(ReleaseType releaseType);
+
+/// {@template release_command}
+/// Creates a new app release for the specified platform(s).
+/// {@endtemplate}
+class ReleaseNewCommand extends ShorebirdCommand {
+  /// {@macro release_command}
+  ReleaseNewCommand({ResolveReleaser? resolveReleaser}) {
+    _resolveReleaser = resolveReleaser ?? getReleaser;
+    argParser
+      ..addOption(
+        'target',
+        abbr: 't',
+        help: 'The main entrypoint file of the application.',
+      )
+      ..addOption(
+        'flavor',
+        help: 'The product flavor to use when building the app.',
+      )
+      ..addOption(
+        'build-number',
+        help: 'The build number of the aar',
+        defaultsTo: '1.0',
+      )
+      ..addOption(
+        'flutter-version',
+        help: 'The Flutter version to use when building the app (e.g: 3.16.3).',
+      )
+      ..addOption(
+        'android-artifact',
+        help:
+            '''The type of artifact to generate. Only relevant for Android releases.''',
+        allowed: ['aab', 'apk'],
+        defaultsTo: 'aab',
+        allowedHelp: {
+          'aab': 'Android App Bundle',
+          'apk': 'Android Package Kit',
+        },
+      )
+      ..addMultiOption(
+        'platform',
+        abbr: 'p',
+        help: 'The platform(s) to to build this release for.',
+        allowed: ReleaseType.values.map((e) => e.cliName).toList(),
+        // TODO(bryanoltman): uncomment this once https://github.com/dart-lang/args/pull/273 lands
+        // mandatory: true.
+      )
+      ..addOption(
+        'release-version',
+        help: '''
+The version of the associated release (e.g. "1.0.0"). This should be the version
+of the iOS app that is using this module.''',
+      )
+      ..addMultiOption(
+        'target-platform',
+        help: 'The target platform(s) for which the app is compiled.',
+        defaultsTo: Arch.values.map((arch) => arch.targetPlatformCliArg),
+        allowed: Arch.values.map((arch) => arch.targetPlatformCliArg),
+      );
+  }
+
+  late final ResolveReleaser _resolveReleaser;
+
+  @override
+  bool get hidden => true;
+
+  @override
+  String get description =>
+      'Creates a shorebird release for the provided target platforms';
+
+  @override
+  String get name => 'release-new';
+
+  @override
+  Future<int> run() async {
+    final releaserFutures = (results['platform'] as List<String>)
+        .map(
+          (platformArg) => ReleaseType.values.firstWhere(
+            (target) => target.cliName == platformArg,
+          ),
+        )
+        .map(_resolveReleaser)
+        .map(createRelease);
+
+    await Future.wait(releaserFutures);
```

Comment:
How will this work with progress logging? Have we tried releasing to multiple platforms in parallel? Also if we fail to release on one platform should that affect the other parallel releasers?

Reply:
We probably want to add some e2e tests to cover this new functionality 🤷

Reply:
Fixed to run in sequence.